### DISCUSSION TITLE
We need development enviroment inside containers.

### DISCUSSION THREAD

**Discussion by bleriot14** [open-webui/open-webui#1923] [General]
**Is your feature request related to a problem? Please describe:**

Hello everyone. My team and I want to be involved in the development of this project. We aim to provide a common environment among developers.

**Describe the solution you'd like:**

For this reason, we want to develop within a containers. Naturally, since we need to run two different services, npm run dev and uvicorn reload, we want to distribute the frontend and backend into two separate containers.  

Has anyone succeeded in this regard?

Thank you.

**Comment by nomppy** [comment]
I am also interested in learning more about this. I would like to make changes in the frontend, e.g. adding SSO, and have changes propagate to the container. Right now it seems like the only way to do this is to rebuild with vite (either manually or with docker compose) and sync the build files into /app/build. Is that really the best way to go? 

If the maintainers can pitch in about how they set up a development environment for Open WebUI, that would be helpful.
